src/hotspot/share/runtime/synchronizer.cpp

```diff
@@ -626,40 +618,15 @@ static intptr_t get_next_hash(Thread* current, oop obj) {
   return value;
 }
 
-static intptr_t install_hash_code(Thread* current, oop obj) {
-  assert(UseObjectMonitorTable, "must be");
-
-  markWord mark = obj->mark_acquire();
-  for (;;) {
-    intptr_t hash = mark.hash();
-    if (hash != 0) {
-      return hash;
-    }
-
-    hash = get_next_hash(current, obj);
-    const markWord old_mark = mark;
-    const markWord new_mark = old_mark.copy_set_hash(hash);
-
-    mark = obj->cas_set_mark(new_mark, old_mark);
-    if (old_mark == mark) {
-      return hash;
-    }
-  }
-}
-
 intptr_t ObjectSynchronizer::FastHashCode(Thread* current, oop obj) {
-  if (UseObjectMonitorTable) {
-    // Since the monitor isn't in the object header, the hash can simply be
-    // installed in the object header.
-    return install_hash_code(current, obj);
-  }
```

Comment:
This merging of the OMT and non-OMT `FastHashCode` does not seem related to the `lightweight` renaming. I found it very hard to convince myself of the equivalences here so would prefer to see this in a separate PR.